app/src/main/java/com/cmput301f20t21/bookfriends/ui/login/CreateAccountActivity.java

```diff
@@ -93,18 +93,85 @@ private boolean checkEmpty(TextInputLayout layout) {
         return isFieldEmpty;
     }
 
+    private boolean checkSpace(TextInputLayout layout) {
+        EditText inputField = layout.getEditText();
+        String userInput = inputField.getText().toString();
+        boolean hasSpace = false;
+        if (userInput.contains(" ")) {
+            layout.setError(getString(R.string.space_error));
+            hasSpace = true;
+        }
+
+        return hasSpace;
+    }
+
+    private boolean checkUsername() {
```

Comment:
I think all these validator functions can go into view model so we can keep our activity class clean. When moving to view model, we can make the function purer by only take in a String param and return boolean. Setting error can still be handled by the activity class.